tests/values.go

```diff
@@ -498,3 +514,89 @@ func ReadValuesAsMap(path string) (map[string]interface{}, error) {
 
 	return values, nil
 }
+
+// ReadValuesAsMap reads the values.yaml from a string.
+func ReadValuesAsMap(valuesStr string) (map[string]interface{}, error) {
+	var values map[string]interface{}
+	decoder := yaml.NewYAMLToJSONDecoder(strings.NewReader(valuesStr))
+	err := decoder.Decode(&values)
+	if err != nil {
+		return nil, fmt.Errorf("error decoding yaml: %w", err)
+	}
+
+	return values, nil
+}
+
+func ReadChartValues(t testing.TB, chart *chart.Chart, valuesStr string) chartutil.Values {
+	valuesMap, err := ReadValuesAsMap(valuesStr)
+	require.NoError(t, err, "read values as map")
+
+	values, err := chartutil.ToRenderValues(chart, valuesMap, DefaultReleaseOptions(), chartutil.DefaultCapabilities.Copy())
+	require.NoError(t, err, "to render values")
+
+	return values
+}
+
+// FindDeployment finds a deployment in the given slice of objects with the
+// given name.
+func FindDeployment(t testing.TB, objs []runtime.Object, name string) *appsv1.Deployment {
```

Comment:
maybe want to return a `(*appsv1.Deployment, bool)` here because there's some test cases where we want to validate that the object doesn't exist (e.g. with `ingress.enable = false`, we want to verify that there is no `Ingress` object)

we can always make that adjustment later though

Reply:
I'll prefix this with a `Must` to leave room for the logic you're describing.